keras/layers/__init__.py

```diff
@@ -1,24 +1,128 @@
+from .merge import Subtract
+from .merge import Multiply
+from .merge import Average
+from .merge import Maximum
+from .merge import Minimum
+from .merge import Concatenate
+from .merge import Dot
+from .merge import add
+from .merge import subtract
+from .merge import multiply
+from .merge import average
+from .merge import maximum
+from .merge import minimum
+from .merge import concatenate
+from .merge import dot
+
+from .core import Dense
+from .core import Activation
+from .core import Dropout
+from .core import Flatten
+from .core import Reshape
+from .core import Permute
+from .core import RepeatVector
+from .core import Lambda
+from .core import ActivityRegularization
+from .core import Masking
+from .core import SpatialDropout1D
+from .core import SpatialDropout2D
+from .core import SpatialDropout3D
+
+from .convolutional import Conv1D
+from .convolutional import Conv2D
+from .convolutional import SeparableConv1D
+from .convolutional import SeparableConv2D
+from .convolutional import DepthwiseConv2D
+from .convolutional import Conv2DTranspose
+from .convolutional import Conv3D
+from .convolutional import Conv3DTranspose
+from .convolutional import Cropping1D
+from .convolutional import Cropping2D
+from .convolutional import Cropping3D
+from .convolutional import UpSampling1D
+from .convolutional import UpSampling2D
+from .convolutional import UpSampling3D
+from .convolutional import ZeroPadding1D
+from .convolutional import ZeroPadding2D
+from .convolutional import ZeroPadding3D
+
+from .pooling import MaxPooling1D
+from .pooling import MaxPooling2D
+from .pooling import MaxPooling3D
+from .pooling import AveragePooling1D
+from .pooling import AveragePooling2D
+from .pooling import AveragePooling3D
+from .pooling import GlobalMaxPooling1D
+from .pooling import GlobalAveragePooling1D
+from .pooling import GlobalMaxPooling2D
+from .pooling import GlobalAveragePooling2D
+from .pooling import GlobalMaxPooling3D
+from .pooling import GlobalAveragePooling3D
+
+from .local import LocallyConnected1D
+from .local import LocallyConnected2D
+
+from .recurrent import RNN
+from .recurrent import SimpleRNN
+from .recurrent import GRU
+from .recurrent import LSTM
+from .recurrent import SimpleRNNCell
+from .recurrent import GRUCell
+from .recurrent import LSTMCell
+
+from .cudnn_recurrent import CuDNNGRU
+from .cudnn_recurrent import CuDNNLSTM
+
+from .normalization import BatchNormalization
+
+from .embeddings import Embedding
+
+from .noise import GaussianNoise
+from .noise import GaussianDropout
+from .noise import AlphaDropout
+
+from .advanced_activations import LeakyReLU
+from .advanced_activations import PReLU
+from .advanced_activations import ELU
+from .advanced_activations import ThresholdedReLU
+from .advanced_activations import Softmax
+from .advanced_activations import ReLU
+
+from .wrappers import Bidirectional
+from .wrappers import TimeDistributed
+
+from .convolutional_recurrent import ConvLSTM2D
+
+# Legacy imports
```

Comment:
A lot of them are current aliases rather than legacy names. I'd suggest co-locating those with the symbol they alias.

Actually legacy objects:

MaxoutDense, Highway, AtrousConvolution1D, AtrousConvolution2D, Recurrent, ConvRecurrent2D